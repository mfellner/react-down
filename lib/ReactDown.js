import React, { PureComponent, PropTypes } from 'react'
import transform from './transform'

export default class ReactDown extends PureComponent {
  props: {
    src: PropTypes.string.isRequired
  }

  render() {
    return transform(this.props.src)
  }
}
