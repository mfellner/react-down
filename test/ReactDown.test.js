import React from 'react'
import renderer from 'react-test-renderer'
import ReactDown from '../lib/ReactDown'

describe('ReactDown', () => {

  it('should render markdown', () => {
    const src = '# Hello, ReactDown!\nThis is *only* a **test**!'
    const component = renderer.create(
      <ReactDown src={src}/>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
