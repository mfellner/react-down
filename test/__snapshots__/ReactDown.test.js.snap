exports[`ReactDown should render markdown 1`] = `
<div>
  <h1>
    Hello, ReactDown!
  </h1>
  <p>
    This is 
    <em>
      only
    </em>
     a 
    <strong>
      test
    </strong>
    !
  </p>
</div>
`;
